noxfile.py

```diff
@@ -93,6 +102,27 @@ def lint(session):
     )
     session.run("flake8", "setup.py", "docs", "dummyserver", "src", "test")
 
+    # TODO: When all files are typed we can change this to .run("mypy", "--strict", "src/urllib3/")
+    session.log("mypy --strict src/urllib3/")
+    for typed_file in TYPED_FILES:
+        if not os.path.isfile(typed_file):
+            session.error("The file {} couldn't be found".format(typed_file))
+        popen = subprocess.Popen(
+            "mypy --strict --follow-imports=skip {}".format(typed_file),
```

Comment:
I realized you just copied what Eland did but why not just use `session.run("mypy", "--strict", "--follow-imports=skip", typed_file)`? cc @sethmlarson 

It seems to work: https://github.com/urllib3/urllib3/commit/1926f5e5279cb680097d706df7e262a0228d587f

Doing that actually finds an issue: `src/urllib3/packages/ssl_match_hostname/_implementation.pyi:3: error: Function is missing a type annotation`

This fixes it: https://github.com/urllib3/urllib3/commit/33c4c13e4d3a5ce77b449b825d8f7214df2e2806

Reply:
TIL about `--follow-imports=skip`, I'll modify Eland to use that option instead.

Reply:
Remembered the reason why not to use it! If you use `--follow-imports=skip` then when you're using something from outside the file then mypy complains that that thing isn't typed in some cases. I'm thinking about switching to something like this on Eland though: 

```python
    errors = []
    popen = subprocess.Popen(
        "mypy --strict eland/",
        env=session.env,
        shell=True,
        stdout=subprocess.PIPE,
        stderr=subprocess.STDOUT,
    )

    mypy_output = ""
    while popen.poll() is None:
        mypy_output += popen.stdout.read(8192).decode()
    mypy_output += popen.stdout.read().decode()

    for line in mypy_output.split("\n"):
        filepath = line.partition(":")[0]
        if filepath in TYPED_FILES:
            errors.append(line)
    if errors:
        session.error("\n" + "\n".join(sorted(set(errors))))
```

Both faster and simpler :)

Reply:
@pquentin Thank you for https://github.com/urllib3/urllib3/commit/33c4c13e4d3a5ce77b449b825d8f7214df2e2806, I was pondering yesterday how do get the types here. May I ask how you got the types? I admit with my previous codebase I was doing `print(type(...))`.

@sethmlarson Great I've incorporated that. Just a quick flag that the `filepath` that are flagged are in fact the stub files, so I've made changes accordingly and renamed `TYPED_FILES` into `STUB_FILES`.

Reply:
> @pquentin Thank you for [33c4c13](https://github.com/urllib3/urllib3/commit/33c4c13e4d3a5ce77b449b825d8f7214df2e2806), I was pondering yesterday how do get the types here. May I ask how you got the types? I admit with my previous codebase I was doing `print(type(...))`.

`match_hostname` is a function that was added to the Python standard library with Python 3.5 and Python 2.7.9. Since urllib3 supported earlier versions of Python 3 at the time and still supports Python <2.7.9 to this day, we just copied its code to urllib3. Knowing that, I simply copied the type hints from https://github.com/python/typeshed/blob/master/stdlib/2and3/ssl.pyi. I included that link in the code, but I should probably have been more explicit about it.

Does that make sense?

However that fix isn't needed since it only applies when mypy is run in the context of nox, and using `subprocess` to run mypy fixes the issue.

@sethmlarson By the way, using `subprocess.run(text=True)` is probably even simpler as "wait for completion and provide all output to a string" will be already provided.